### PR TITLE
Add option to change user before continuing OAuth flow

### PR DESCRIPTION
**Pull request by hossenlopp**
- Overrides doorkeeper operation to always ask user
- Adds change user button
- After user logs back in they are returned to the auth flow
- Bonnie branding and styles are now used in the auth request page

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1741
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases N/A Doorkeeper overrides are unable to be tested.
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) )
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: **Doorkeeper override code cannot be easily tested.**
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 

Branch | Back End Coverage | Front End Coverage
-- | -- | --
bonnie-v3.0 | 2164 / 2415 LOC (89.61%) covered | [Paste Screenshot Here]
[New Branch] | 2166 / 2418 LOC (89.58%) covered | [Paste Screenshot Here]

- [x] Automated regression test(s) pass N/A does not affect calculation

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA test links:
- [x] Justification for using JIRA tests:
- [x] JIRA tests have been added to sprint

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/controllers/doorkeeper_override/authorizations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    # of the application.
+    def new
+      if pre_auth.authorizable?
+        render :new
+      else
+        render :error
+      end
+    end
+
+    # Additional functional to support change user button. This logs the user out, stores the page the url
```
Comment (jbradl11) [inline app/controllers/doorkeeper_override/authorizations_controller.rb]: *functionality* ?

Comment (hossenlopp) [reply app/controllers/doorkeeper_override/authorizations_controller.rb]: Yep... I'll fix that.

**Review by daco101 — COMMENTED** [review]
One rubocop error, otherwise looks good to me.

**Review by jbradl11 — APPROVED** [review]
Looks good. Great feature.

**Review by daco101 — APPROVED** [review]

**Pull request merged**